pkg/alertmanager/multitenant.go

```diff
@@ -117,6 +118,7 @@ func (cfg *MultitenantAlertmanagerConfig) RegisterFlags(f *flag.FlagSet) {
 
 	f.StringVar(&cfg.ClusterBindAddr, "cluster.listen-address", defaultClusterAddr, "Listen address for cluster.")
 	f.StringVar(&cfg.ClusterAdvertiseAddr, "cluster.advertise-address", "", "Explicit address to advertise in cluster.")
+	f.DurationVar(&cfg.ClusterGossipInterval, "cluster.gossip-interval", cluster.DefaultGossipInterval, "The interval between sending messages that need to be gossiped that haven't been able to piggyback on probing messages. If this is set to zero, non-piggyback gossip is disabled. By lowering this value (more frequent) gossip messages are propagated across cluster more quickly at the expense of increased bandwidth.")
```

Comment:
Not too sure why we are not using an `-alertmanager.` prefix for all of those config flags, as currently it might mislead users to think it is affecting other parts of the system to (e.g. memberlist)

Reply:
What is non-piggyback gossip?

If we allow configuring this value (which I think is already pretty low for gossip, and wouldn't recommend going to lower values), we should also enable configuration of pull/push interval for faster state convergence. (In addition to that, configuring GossipNodes would be useful too, but that doesn't seem configurable in AM).

Reply:
> Not too sure why we are not using an -alertmanager. prefix for all of those config flags, as currently it might mislead users to think it is affecting other parts of the system to (e.g. memberlist)

I think it's a bug and we should fix it (but need to address it in a separate PR). I've opened an issue in the meanwhile: https://github.com/cortexproject/cortex/issues/3670

Reply:
I have filed https://github.com/cortexproject/cortex/pull/3677 to fix #3670. Would be great to align these two PRs to make sure the nomenclature and prefix make sense on both. Happy to merge this as is and I can quickly fix it the other.

Reply:
> What is non-piggyback gossip?

That was a copy paste from the memberlist code comment. I think it is confusing more than helping so I reverted to the alertmanager description.

> pull/push interval 

Added that

> Gossip Nodes

Yes that seems to be fixed at quorum or 3 whatever is bigger: https://github.com/prometheus/alertmanager/blob/ce108378d4c8d580804452dcc3a31222067793b2/cluster/cluster.go#L184-L187 

> Aliging to #3677 

I have added some comments [here] (but meanwhile I used HA in the same way)

[here]:https://github.com/cortexproject/cortex/pull/3677/files#r555644111